aries_cloudagent/vc/vc_di/cryptosuites/eddsa_jcs_2022.py

```diff
@@ -0,0 +1,119 @@
+"""EddsaJcs2022 cryptosuite."""
+
+from hashlib import sha256
+import canonicaljson
+import nacl
+
+from ....wallet.base import BaseWallet
+from ....utils.multiformats import multibase
+from ....core.profile import Profile
+from .. import DataIntegrityProofException
+
+
+class EddsaJcs2022:
+    """EddsaJcs2022 suite."""
+
+    def __init__(self, *, profile: Profile):
+        """Create new EddsaJcs2022 instance.
+
+        Args:
+            profile: Key profile to use.
+        """
+        super().__init__()
+        self.profile = profile
+
+    async def _serialization(self, hash_data, options):
+        # https://www.w3.org/TR/vc-di-eddsa/#proof-serialization-eddsa-jcs-2022
+        async with self.profile.session() as session:
+            did_info = await session.inject(BaseWallet).get_local_did(
+                options["verificationMethod"].split("#")[0]
+            )
+        async with self.profile.session() as session:
+            wallet = session.inject(BaseWallet)
+        proof_bytes = await wallet.sign_message(
+            message=hash_data,
+            from_verkey=did_info.verkey,
+        )
+        return proof_bytes
+
+    async def add_proof(self, document, proof_options):
+        # https://www.w3.org/TR/vc-data-integrity/#add-proof
+        """Verify the data against the proof.
+
+        Args:
+            document: The data to sign.
+            proof_options: The proof options.
+
+        Returns:
+            verification_response: Whether the signature is valid for the data
+
+        """
+
+        existing_proof = document.pop("proof", [])
+        assert isinstance(existing_proof, list) or isinstance(existing_proof, dict)
+        existing_proof = (
+            [existing_proof] if isinstance(existing_proof, dict) else existing_proof
+        )
+
+        assert proof_options["type"] == "DataIntegrityProof"
+        assert proof_options["cryptosuite"] == "eddsa-jcs-2022"
+        assert proof_options["proofPurpose"]
+        assert proof_options["verificationMethod"]
+
+        try:
+            hash_data = (
+                sha256(canonicaljson.encode_canonical_json(document)).digest()
+                + sha256(canonicaljson.encode_canonical_json(proof_options)).digest()
+            )
+            proof_bytes = await self._serialization(hash_data, proof_options)
+
+            proof = proof_options.copy()
+            proof["proofValue"] = multibase.encode(proof_bytes, "base58btc")
+
+            secured_document = document.copy()
+            secured_document["proof"] = existing_proof
+            secured_document["proof"].append(proof)
+
+            return secured_document
+        except Exception:
+            raise DataIntegrityProofException()
+
+    async def verify_proof(self, unsecured_document, proof):
+        # https://www.w3.org/TR/vc-data-integrity/#verify-proof
```

Comment:
Nitpick: why not put these links (which are very helpful) into the docstring? This would ensure that they are rendered into the documentation.

Reply:
will do, thanks for the tip (I think ruff was giving me warning when I was doing so but I'll revisit it)